pkg/store/proxy.go

```diff
@@ -186,12 +188,35 @@ func WithProxyStoreMatcherConverter(mc *storepb.MatcherConverter) ProxyStoreOpti
 }
 
 // WithBlockedMetricPatterns returns a ProxyStoreOption that sets the blocked metric patterns.
+// It parses input patterns to extract prefixes (like "kube_", "envoy_") using regex
+// and stores them in a radix tree for efficient prefix matching. Exact patterns
+// (like "up") are stored in a set for whole match checking.
 func WithBlockedMetricPatterns(patterns []string) ProxyStoreOption {
 	return func(s *ProxyStore) {
-		s.blockedMetricPatterns = radix.New()
+		s.blockedMetricPrefixes = radix.New()
+		s.blockedMetricExacts = make(map[string]struct{})
+
+		// Regex to match patterns ending with * (wildcard patterns)
+		starPattern := regexp.MustCompile(`^(.+)\*$`)
+		// Regex to match patterns ending with _ (underscore prefix patterns)
+		underscorePattern := regexp.MustCompile(`^([a-zA-Z_][a-zA-Z0-9_]*_)$`)
```

Comment:
good assumption all pattern ends with `_` is also only prefix match. in this case i would just extract last letter to see if it is `*` or `_`